AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -169,6 +171,19 @@ - (void)setDelegate:(id<ASNetworkImageNodeDelegate>)delegate
 - (void)displayWillStart
 {
   [super displayWillStart];
+  
+  if (_cacheSupportsSynchronousFetch) {
+    {
+      ASDN::MutexLocker l(_lock);
+      if (_URL && _downloadIdentifier == nil) {
+        UIImage *result = [_cache synchronouslyFetchedCachedImageWithURL:_URL];
```

Comment:
As long as PINRemoteImage is only used with ASDK, yes. Basically, if you mix shouldSkipDecode and not skipping decode, you can't tell what is going to end up in the cache (I want to address this in PINRemoteImage). In the default case though, it should always be uncompressed.